contracts/modules/STO/POLYCappedSTO.sol

```diff
@@ -0,0 +1,526 @@
+     * @param _nonAccreditedLimit Array of uints specifying non-accredited limits
+     */
+    function changeNonAccreditedLimit(address[] memory _investors, uint256[] memory _nonAccreditedLimit) public onlyOwner {
+        require(_investors.length == _nonAccreditedLimit.length, "Length mismatch");
+        for (uint256 i = 0; i < _investors.length; i++) {
+            nonAccreditedLimitOverride[_investors[i]] = _nonAccreditedLimit[i];
+            emit SetNonAccreditedLimit(_investors[i], _nonAccreditedLimit[i]);
+        }
+    }
+
+    /**
+     * @notice Function to set allowBeneficialInvestments (allow beneficiary to be different to funder)
+     * @param _allowBeneficialInvestments Boolean to allow or disallow beneficial investments
+     */
+    function changeAllowBeneficialInvestments(bool _allowBeneficialInvestments) external onlyOwner {
+        require(_allowBeneficialInvestments != allowBeneficialInvestments, "Does not change value");
+        allowBeneficialInvestments = _allowBeneficialInvestments;
+        emit SetAllowBeneficialInvestments(allowBeneficialInvestments);
+    }
+
+    //////////////////////////
+    // Investment Functions //
+    //////////////////////////
+
+    /**
+    * @notice no fallback function - non-payable
+    function() external {}
+    */
+
+    /**
+      * @notice Purchase tokens using POLY
+      * @param _beneficiary Address where security tokens will be sent
+      * @param _investedPOLY Amount of POLY invested
+      */
+    function buyWithPOLY(address _beneficiary, uint256 _investedPOLY) external {
+        _buyWithTokens(_beneficiary, _investedPOLY, FundRaiseType.POLY, polyToken);
+    }
+
+    function _buyWithTokens(
+        address _beneficiary,
+        uint256 _tokenAmount,
+        FundRaiseType _fundRaiseType,
+        IERC20 _token
+    ) internal {
+        if (!allowBeneficialInvestments) {
+            require(_beneficiary == msg.sender, "Beneficiary != msg.sender");
+        }
+        require(_canBuy(_beneficiary), "Unauthorized");
+        uint256 _spentValue = _buyTokens(_beneficiary, _tokenAmount, _fundRaiseType);
+        // Forward coins to issuer wallet
+        require(_token.transferFrom(msg.sender, wallet, _spentValue), "Transfer failed");
+        emit FundsReceived(msg.sender, _beneficiary, _fundRaiseType, _tokenAmount, _spentValue);
+        _postValidatePurchase(_beneficiary, _spentValue);
+    }
+
+    /**
+      * @notice Low level token purchase
+      * @param _beneficiary Address where security tokens will be sent
+      * @param _investmentValue Amount of POLY invested
+      * @param _fundRaiseType Fund raise type (POLY)
+      */
+    function _buyTokens(
+        address _beneficiary,
+        uint256 _investmentValue,
+        FundRaiseType _fundRaiseType
+    )
+        internal
+        nonReentrant
+        whenNotPaused
+        returns(uint256 _spentValue)
+    {
+        uint256 _tokens;
```

Comment:
`_` should be used as a prefix for only the variables that are function parameters.